packages/cypher-builder/src/procedures/Yield.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Clause } from "../clauses/Clause";
+import { WithReturn } from "../clauses/mixins/WithReturn";
+import { WithWhere } from "../clauses/mixins/WithWhere";
+import { WithWith } from "../clauses/mixins/WithWith";
+import type { ProjectionColumn } from "../clauses/sub-clauses/Projection";
+import { Projection } from "../clauses/sub-clauses/Projection";
+import { mixin } from "../clauses/utils/mixin";
+import type { CypherEnvironment } from "../Environment";
+import type { Literal } from "../references/Literal";
+import type { Variable } from "../references/Variable";
+import { NamedVariable } from "../references/Variable";
+import type { Expr } from "../types";
+import { asArray } from "../utils/as-array";
+import { compileCypherIfExists } from "../utils/compile-cypher-if-exists";
+
+export type YieldProjectionColumn<T extends string> = T | [T, Variable | Literal];
+
+export interface Yield extends WithReturn, WithWhere, WithWith {}
+
+/** Yield statement after a Procedure CALL
+ * @see [Cypher Documentation](https://neo4j.com/docs/cypher-manual/current/clauses/call/#call-call-a-procedure-call-yield-star)
+ * @group Expressions
+ * @category Procedures
+ */
+@mixin(WithReturn, WithWhere, WithWith)
+export class Yield extends Clause {
```

Comment:
No blocking, but I believe we should treat `YIELD` as a sub-clause rather than a clause.

Reply:
That's a good point, I think that makes sense, I'll take a look at that